userspace/libsinsp/parsers.cpp

```diff
@@ -2247,6 +2253,58 @@ inline void sinsp_parser::add_socket(sinsp_evt *evt, int64_t fd, uint32_t domain
 	evt->m_fdinfo = evt->m_tinfo->add_fd(fd, &fdi);
 }
 
+/**
+ * If we receive a call to 'sendto()' and the event's m_fdinfo is nullptr,
+ * then we likely missed the call to 'socket()' that created the file
+ * descriptor.  In that case, we'll guess that it's a SOCK_DGRAM/UDP socket
+ * and create the fdinfo based on that.
+ *
+ * Precondition: evt->m_fdinfo == nullptr
+ */
+inline void sinsp_parser::infer_sendto_fdinfo(sinsp_evt* const evt)
+{
+	ASSERT(evt->m_fdinfo == nullptr);
+	ASSERT(evt->m_tinfo != nullptr);
```

Comment:
Do we want an assert here or an early return? I'm not sure we can rely 100% on tinfo being always there

Reply:
I consider those two conditions to be preconditions for the method; the caller is responsible for ensuring that those conditions hold before calling the function.  In that case, I think `ASSERT` is reasonable, because if it fires it indicates that the method is getting misused and the caller needs to be fixed (as opposed to silently ignoring the error).

That said, as far as I know `ASSERT` is defined meaningful only for debug builds.  It's not clear to me if/when we ever actually run debug builds, so maybe `ASSERT` isn't buying us much.